Objects/genericaliasobject.c

```diff
@@ -254,15 +285,46 @@ ga_getitem(PyObject *self, PyObject *item)
         if (typevar) {
             Py_ssize_t iparam = tuple_index(alias->parameters, nparams, arg);
             assert(iparam >= 0);
-            if (is_tuple) {
-                arg = PyTuple_GET_ITEM(item, iparam);
+            arg = args[iparam];
+            Py_INCREF(arg);
+        }
+        else {
+            _Py_IDENTIFIER(__parameters__);
+            PyObject *subparams;
+            if (_PyObject_LookupAttrId(arg, &PyId___parameters__, &subparams) < 0) {
+                Py_DECREF(newargs);
+                return NULL;
+            }
+            if (subparams && PyTuple_Check(subparams)) {
+                Py_ssize_t nsubargs = PyTuple_GET_SIZE(subparams);
```

Comment:
Again, you can reuse `arg` if `nsubargs` is zero.

Reply:
Good catch. There was a bug (`dict[T, list[int]][str]` did not work), fixed and added a test.